main.py

```diff
@@ -192,9 +195,12 @@ def main():
 
                 # evaluate
                 if total_steps % args.eval_freq == 0:
-                    eval_reward = evaluate_policy(gac, eval_env, args.eval_episodes)
+                    eval_rewards = evaluate_policy(gac, eval_env, args.eval_episodes)
+                    eval_reward = sum(eval_rewards) / args.eval_episodes
                     print('eval_reward:', eval_reward)
-                    results_dict['eval_rewards'].append((total_steps, eval_reward))
+                    results_dict['eval_rewards'].append((total_steps, eval_rewards))
+                    with open ('results.txt', 'w') as file:

```

Comment:
This will add substantial overhead if run in the loop. Just run this write operation after the end of the training loop before the save_models function.

Reply:
No, this is what I insist. I would like to save the results every time we evaluate. It's not "substantial" compared with training. The reason is, even we kill the job, we can still have data there to plot.